packages/wdio-mocha-framework/src/index.js

```diff
@@ -245,6 +247,11 @@ class MochaAdapter {
                 message.passed = (params.payload.state === 'passed')
             }
 
+            if (params.payload.parent?.title && mochaAllHooksIfPresent) {
+                const hookName = mochaAllHooksIfPresent[0]
+                message.title = `${hookName} for ${params.payload.parent.title}`
```

Comment:
Do you think we need "for ${params.payload.parent.title}`"?

Reply:
As per the comments on the original issue, the ask was `beforeAll` & `afterAll` mocha hook should take the parent test title while showing up in reports. That's why I have included  `params.payload.parent.title` in the message title. Otherwise the title would show just `before all hook` in the report.